src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterEngine.java

```diff
@@ -116,15 +117,17 @@ public String getModelId(Map<String, Object> options) {
      * @param modelId The ID of the model to use
      * @param question The search query
      * @param context The document text
+     * @param preTag The pre tag to use for highlighting
+     * @param postTag The post tag to use for highlighting
      * @return Formatted text with highlighting
      */
-    public String getHighlightedSentences(String modelId, String question, String context) {
+    public String getHighlightedSentences(String modelId, String question, String context, String preTag, String postTag) {
```

Comment:
nit: Can we create a DTO object for the arguments that contains all the variables required for highlighting. Something around HighlighterDTO. Because later when this feature will be enabled for other queries like Hybrid we can reuse the DTO object.

Reply:
As of now let's keep the current parameter-based approach since it's simpler. We can revisit using a DTO if it becomes more complex. Actually hybrid type query is supported in the semantic highlighter, it's not related to this part code.